demos/Animation/main.py

```diff
@@ -0,0 +1,64 @@
+import gi
+
+gi.require_version("Gtk", "4.0")
+gi.require_version("Adw", "1")
+
+from gi.repository import Gtk, Gsk, Graphene, Adw
+import workbench
+
+
+button_timed = workbench.builder.get_object("button_timed")
+progress_bar = workbench.builder.get_object("progress_bar")
+target_timed = Adw.PropertyAnimationTarget.new(progress_bar, "fraction")
+
+animation_timed = Adw.TimedAnimation(
+    widget=progress_bar,
+    value_from=0,
+    value_to=1,
+    duration=1500,
+    easing=Adw.Easing.EASE_IN_OUT_CUBIC,
+    target=target_timed,
+)
+
+button_timed.connect("clicked", lambda _: animation_timed.play())
+animation_timed.connect("done", lambda _: animation_timed.reset)
+
+button_spring = workbench.builder.get_object("button_spring")
+ball = workbench.builder.get_object("ball")
+
+
+def get_end_position(widget):
+    parent_width = widget.get_parent().get_width()
+    width = ball.get_width()
+
+    return parent_width - width
+
+
+def move_widget(widget, x, y):
+    transform = Gsk.Transform.new()
+    p = Graphene.Point.new(x=x, y=y)
+    transform = transform.translate(p)
+    widget.allocate(widget.get_width(), widget.get_height(), -1, transform)
+
+
+def animation_cb(value):
+    end = get_end_position(ball)
+    x = Adw.lerp(0, end, value)
+    move_widget(ball, x, 0)
+
+
+target_spring = Adw.CallbackAnimationTarget.new(animation_cb)
+params = Adw.SpringParams.new(
+    damping_ratio=0.5,
+    mass=1.0,
+    stiffness=50.0,
+)
+animation_spring = Adw.SpringAnimation(
+    widget=ball, value_from=0, value_to=1, spring_params=params, target=target_spring
+)
+animation_spring.initial_velocity = 1.0
+# If amplitude of oscillation < epsilon, animation stops
+animation_spring.epsilon = 0.001
+animation_spring.clamp = False
```

Comment:
These properties must be accessed either through `animation_spring.props` or using the setters.
```suggestion
animation_spring.set_initial_velocity(1.0)
# If amplitude of oscillation < epsilon, animation stops
animation_spring.set_epsilon(0.001)
animation_spring.set_clamp(False)
```

Reply:
Are you talking about "Elastic" the AI search engine? Haven't heard of it before. As you see it uses e.g. `animation.props.epsilon` and not `animation.epsilon`. As I said properties must be accessed either through the `props` namespace (like suggested by Elastic in the example you gave) or through the setter (`animation.set_epsilon(0.00315)`). In contrast the attribute `animation.epsilon` does not exist. You can write to it, but it will not affect the result of the getter `animation.get_epsilon()`, so it doesn't really do what you want.